planet/clients/orders.py

```diff
@@ -271,18 +273,41 @@ async def download_asset(self,
                              progress_bar=progress_bar)
         return dl_path
 
+    @staticmethod # should this be an async function?
+    def calculate_checksum(origin_hash: str,
+                        filename: str,
+                        checksum: str):
+        """Calculate checksum and validate that it passes."""
+        if checksum == 'MD5':
```

Comment:
I think you should only compare `checksum` to strings without a care of its case (i.e., uppercase or lowercase). Here this fails because on line 351 you compare to `checksum.lower()`, which is great! (i.e., `origin_hash = json_entry['digests'][checksum.lower()]`). Let's either do `checksum.lower()` everywhere, or maybe ever more robust, we can assign `checksum = checksum.lower()` at the beginning and now you know `md5` and `sha256` will always be lowercase. Or, another option would be to compare without caring about cases, which would be like `checksum.casefold() == 'MD5'.casefold()`.